examples/mysensor/README.md

```diff
@@ -2,7 +2,52 @@
 
 This example demonstrates a user defining a new sensor model.
 
+
+## How to add custom models
+
+Models can be added to the viam-server by creating a struct that implements the interface of the subtype we want to register.
+For example, if we want to create a sensor, we would want to make sure that we implement the Sensor interface, which includes the method `Readings` and also `DoCommand` from the `generic.Generic` interface.
+
+```
+    type Sensor interface {
+        // Readings return data specific to the type of sensor and can be of any type.
+        Readings(ctx context.Context) (map[string]interface{}, error)
+        generic.Generic
+    }
+```
+
+To make sure the struct we create implements the interface, we can check it using the go linter like this.
+```
+    var _ = sensor.Sensor(&mySensor{})
+```
+
+The model then has to be registered.
+
+```
+    // registering the component model on init is how we make sure the new model is picked up and usable.
+    func init() {
```

Comment:
it'd be good to be explicit about where this should live. is it the file implementing the new sensor or somewhere else?

Reply:
added